### PR TITLE
Add @jurajseffer work on waiting for status change

### PR DESCRIPTION
**Pull request by hylaride**
##### SUMMARY
If it's not considered rude, I'm creating a PR with changes from @jurajseffer so that this very convenient feature can get in.

It allows the module to wait until a specific status change on the ELB target group has happened (eg fully added/drained) before continuing.

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME
elb_target module

##### ANSIBLE VERSION
2.4.0.0

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by jurajseffer** [comment]
Thank you Christopher, I've been too busy using it to get the PR done! :) :+1:

**Comment by hylaride** [comment]
@wimnat How does this PR look?

File: lib/ansible/modules/cloud/amazon/elb_target.py [hunk 1]
```diff
@@ -185,19 +202,35 @@ def deregister_target(connection, module):
...
+    if 'State' in target_description['TargetHealth']:
+        if target_description['TargetHealth']['State'] not in ['draining', 'unused']:
+            try:
+                connection.deregister_targets(TargetGroupArn=target_group_arn, Targets=[target])
+                changed = True
+                if target_status:
+                    target_status_check(connection, module, target_group_arn, target, target_status, target_status_timeout)
+
+            except ClientError as e:
+                module.fail_json(msg="Ohh parps", exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
```
Comment (wimnat) [inline lib/ansible/modules/cloud/amazon/elb_target.py]: Better message here than "Ohh parps"? ;)

File: lib/ansible/modules/cloud/amazon/elb_target.py [hunk 2]
```diff
@@ -207,6 +240,8 @@ def main():
             target_group_name=dict(type='str'),
             target_id=dict(type='str', required=True),
             target_port=dict(type='int'),
+            target_status=dict(type='str'),
```
Comment (wimnat) [inline lib/ansible/modules/cloud/amazon/elb_target.py]: Can you provide the set of statuses here as a choices list?

Comment (hylaride) [reply lib/ansible/modules/cloud/amazon/elb_target.py]: Done!

File: lib/ansible/modules/cloud/amazon/elb_target.py (hunk 1 above)
Comment (hylaride) [reply lib/ansible/modules/cloud/amazon/elb_target.py]: Set to use `e.message` 💯

**Comment by hylaride** [comment]
@wimnat Updated with your requests! 😃

**Comment by jurajseffer** [comment]
@hylaride Thanks for doing all the legwork :D Perhaps `unavailable` should be added to the possible statuses for completness - my guess is that it means ec2 instance/target doesn't even exist. As per http://docs.aws.amazon.com/elasticloadbalancing/latest/APIReference/API_TargetHealth.html

**Comment by hylaride** [comment]
@jurajseffer AH! I was basing it on [these docs](http://docs.aws.amazon.com/elasticloadbalancing/latest/application/target-group-health-checks.html) (under `Target Health Status`), but I think that's just what the AWS console shows.  I'll add `unavailable` as this is via the API.

**Comment by hylaride** [comment]
@jurajseffer And Thank you for the original code!

**Comment by hylaride** [comment]
@wimnat How do things look now?

**Pull request merged**